legacy/builder/setup_build_properties.go

```diff
@@ -126,6 +127,14 @@ func (s *SetupBuildProperties) Run(ctx *types.Context) error {
 
 	buildProperties.Merge(ctx.PackageManager.CustomGlobalProperties)
 
+	keychainProp := buildProperties.ContainsKey("build.keys.keychain")
+	signProp := buildProperties.ContainsKey("build.keys.sign_key")
+	encryptProp := buildProperties.ContainsKey("build.keys.encrypt_key")
+	// we verify that all the properties for the secure boot keys are defined or none of them is defined.
+	if !(keychainProp || signProp || encryptProp) && (keychainProp && signProp && encryptProp) {
+		return errors.Errorf("%s core does not specify correctly default sign and encryption keys", ctx.BuildCore)
```

Comment:
```suggestion
		return errors.Errorf("%s platform does not specify correctly default sign and encryption keys", targetPlatform.Platform)
```
The previous error message was incorrect because the build properties are not defined by the [core](https://arduino.github.io/arduino-cli/dev/platform-specification/#cores).

The previous error message was also prone to ambiguity. It produced this error message in my test:

```text
Error during build: arduino core does not specify correctly default sign and encryption keys
```

You can see the core of the platform I tested with is named `arduino` (as is the case with most of them):
https://github.com/arduino/ArduinoCore-mbed/tree/master/cores/arduino

Since the check is done on all installed platforms, rather than only the board being compiled for, this would leave the user wondering which of their platforms was being referred to. The platform identifier leaves no doubt:

```text
Error during build: arduino:mbed_portenta platform does not specify correctly default sign and encryption keys
```